enclave/src/attestation.rs

```diff
@@ -533,19 +453,7 @@ pub fn create_attestation_report(
         return Err(sgx_status_t::SGX_ERROR_UNEXPECTED);
     }
 
-    let quote_vec: Vec<u8> = return_quote_buf[..quote_len as usize].to_vec();
-    let res =
-        unsafe { ocall_get_ias_socket(&mut rt as *mut sgx_status_t, &mut ias_sock as *mut i32) };
-
-    if res != sgx_status_t::SGX_SUCCESS {
-        return Err(res);
-    }
-
-    if rt != sgx_status_t::SGX_SUCCESS {
-        return Err(rt);
-    }
-
-    let (attn_report, sig, cert) = get_report_from_intel(ias_sock, quote_vec)?;
+    let (attn_report, sig, cert) = get_report_from_intel(ias_socket, quote_content)?;
```

Comment:
This was a bit confusing to me, we're getting the `ias_socket` a second time in the same function. Is this really needed? Can't we just re-use the socket from the first call? Or are there some side-effects such that the second call will return a different socket number?

Reply:
Hmm, I don't really know. This is basically copied sample code from the teaclaves repository.

If we'd want to find out, I believe we just need to test that.

Reply:
Ah okay - can you point me to where in the teaclave repository this code resides?

Reply:
This is the corresponding file: https://github.com/apache/incubator-teaclave-sgx-sdk/blob/master/samplecode/mutual-ra/enclave/src/lib.rs